### PR TITLE
20230208-freertos-update

### PR DESCRIPTION
**Pull request by douzzer**
`Makefile`: add `HOST=` and `RUNTIME=` make options, e.g. HOST=arm-none-eabi RUNTIME=FreeRTOS-lwIP;

`Makefile.analyzers`: put "passed: " at start of every pass message; add --ignore-fail-on-non-empty to rmdirs; add no-getprotoby-test; add -`DWOLFSENTRY_NO_GETPROTOBY` to minimal-build-test;

`scripts/build_wolfsentry_options_h.awk`: add support for -U arguments;

wolfsentry/wolfsentry.h: change long to `time_t` for args in seconds of `wolfsentry_to_epoch_time_cb_t`, `wolfsentry_from_epoch_time_cb_t`, `wolfsentry_interval_to_seconds_cb_t`, `wolfsentry_interval_from_seconds_cb_t`, and `wolfsentry_set_deadline_abs()`; add `_wolfsentry_get_n_mallocs()` ifdef `WOLFSENTRY_MALLOC_DEBUG`;

`wolfsentry/wolfsentry_errcodes.h`: add some convenience provisions for embedded targets, e.g. `WOLFSENTRY_ERROR_DECODE_ERROR_CODE()` and friends return ints now; refactor `WOLFSENTRY_ERROR_RETURN_1()` and friends to use `WOLFSENTRY_PRINTF_ERR()`; add new code `WOLFSENTRY_SUCCESS_ID_USED_FALLBACK`;

`wolfsentry/wolfsentry_settings.h`: add various `FREERTOS` defaults; eliminate `enumint_t` type (not portable); refactor `WOLFSENTRY_THREADSAFE` setup; change struct `wolfsentry_thread_context_public` to be an array of `uint64_t`'s instead of pointers, for portability;

`wolfsentry/wolfsentry_util.h`: add `attr_align_to()`;

`src/wolfsentry_internal.h`: move much of the FreeRTOS setup to `wolfsentry_settings.h`; separate and name ("`wolfsentry_rwlock_state`") the enum used for `wolfsentry_rwlock.state` and make the latter a `uint32_t`; change use of `WOLFSENTRY_THREAD_GET_ID_HANDLER` to not expect () in the definition; add `attr_align_to(8)` to `wolfsentry_table_ent_header` for portability;

`src/json/load_config.c`: refactor out `enumint_t`; add support for `WOLFSENTRY_NO_GETPROTOBY` gate; refactor `snprintf()` args for portability to newlib-nano;

`src/routes.c`: add support for `WOLFSENTRY_NO_GETPROTOBY` gate;

`src/wolfsentry_util.c`: add a `WOLFSENTRY_CALL_DEPTH_RETURNS_STRING` path to `_wolfsentry_call_depth()` for use with newlib-nano; add `WOLFSENTRY_MALLOC_DEBUG` code paths in `WOLFSENTRY_MALLOC_BUILTINS` implementations; add `WOLFSENTRY_MALLOC_BUILTINS` implementation for native FreeRTOS heap, including `wolfsentry_builtin_realloc()`; add `freertos_now()`; add a fallback ID assignment path to `wolfsentry_init_thread_context()` if the OS lookup fails; add check for `WOLFSENTRY_THREAD_NO_ID` to thread and lock functions; remove unused FreeRTOS implementation `UTILS_NanosecondsToTimespec()`; refactor `freertos_sem_timedwait()` to use `freertos_now()`; fix return value bug in `wolfsentry_lock_unlock()`; add `WOLFSENTRY_ERROR_RERETURN()` wrappers to `wolfsentry_context_lock_*()`; refactor FreeRTOS `wolfsentry_builtin_get_time()` to use `freertos_now()`; fix math around `config->route_private_data_size` and `config->route_private_data_alignment` in `wolfsentry_eventconfig_check()` and `wolfsentry_eventconfig_load()`.

### CODE REVIEW

File: FreeRTOS/include/FreeRTOSConfig.h [hunk 1]
```diff
@@ -0,0 +1,129 @@
+/*
+ * FreeRTOS V202212.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
```
Comment (dgarske) [inline FreeRTOS/include/FreeRTOSConfig.h:3]: Please fix license headers. Can't be this, must be ours. If can't be fixed then tell user how to get these or move to examples repo.

File: Makefile [hunk 2]
```diff
@@ -50,6 +50,42 @@ ifndef OPTIM
...
+    endif
+    ifeq "$(AS)" "as"
+        AS = $(HOST)-as
+    endif
+endif
+
+ifdef RUNTIME
+    ifeq "$(RUNTIME)" "FreeRTOS-lwIP"
+        ifndef FREERTOS_TOP
+            $(error FREERTOS_TOP not supplied with RUNTIME=$(RUNTIME))
```
Comment (dgarske) [inline Makefile:77]: Getting this: 
```
make HOST=arm-none-eabi RUNTIME=FreeRTOS-lwIP
Makefile:77: *** FREERTOS_TOP not supplied with RUNTIME=FreeRTOS-lwIP.  Stop.
```
But don't see `FREERTOS_TOP` mentioned in PR comments or in any files.

**Review by dgarske — CHANGES_REQUESTED** [review]
Are you planning to supply any type of README.md for how to build with FreeRTOS/LWIP?

File: FreeRTOS/include/FreeRTOSConfig.h (hunk 1 above)
Comment (douzzer) [reply FreeRTOS/include/FreeRTOSConfig.h:3]: will do.  added those last-minute last night to get the bundle to build.

File: Makefile (hunk 2 above)
Comment (douzzer) [reply Makefile:77]: will fix.  I had sent some out-of-band build instructions, ```make -j HOST=arm-none-eabi RUNTIME=FreeRTOS-lwIP FREERTOS_TOP=../third/FreeRTOSv202212.00/FreeRTOS/Source LWIP_TOP=../third/lwip/src EXTRA_CFLAGS='-mthumb -mcpu=cortex-m7'``` but that needs to make it into the distribution.  I'll add documentation around it in README.md, and I'm also adding a subtest to `make check` to build+unittest under arm-none-eabi-FreeRTOS-lwIP.

**Review by dgarske — APPROVED** [review]

**Pull request merged**